src/corelibs/U2Gui/src/util/project/ProjectTreeController.cpp

```diff
@@ -792,8 +795,64 @@ void ProjectTreeController::sl_onRemoveSelectedItems() {
     const QList<Document *> selectedDocs = getDocsInSelection(deriveDocsFromObjs).values();
     const QList<Folder> selectedFolders = getSelectedFolders();
     const QList<GObject *> selectedObjects = objectSelection.getSelectedObjects();
+    
+    QStringList selectedItemNames;
+    foreach(Document * doc, selectedDocs) {
+        selectedItemNames.append(doc->getName());
+    }
+    foreach(const Folder& folder, selectedFolders) {
+        selectedItemNames.append(folder.getFolderName());
+    }
+    foreach(GObject* obj, selectedObjects) {
+        selectedItemNames.append(obj->getGObjectName());
+    }
+    
+    QString fullItemsNamesList;
+
+    QString warningMessageText = (1 == selectedItemNames.count()) ? tr("Remove selected item?") : tr("Remove selected items?");
+    warningMessageText += "<ul style=\"margin-top:5px;margin-bottom:0px\"><li>";
+
+    const bool tooManyItemsSelected = (MAX_DISPLAING_NAME_COUNT < selectedItemNames.count());
 
-    removeItems(selectedDocs, selectedFolders, selectedObjects);
+    int itemCounter = 0;
+    foreach (QString name, selectedItemNames) {
+        if (tooManyItemsSelected) {
+            fullItemsNamesList += " - " + name + "\n";
+        }
+
+        if (MAX_DISPLAING_NAME_COUNT >= ++itemCounter) {
+            // cut long names
+            if (NAME_DISPLAYING_SYMBOLS_COUNT < name.size()) {
+                name = name.left(NAME_DISPLAYING_SYMBOLS_COUNT);
+                name += "...";
+            }
+            warningMessageText += name;
+            warningMessageText += "</li><li>";
```

Comment:
break from the loop here? Why do you want to add dozens of <li>...</li> elements here?